### PR TITLE
Documentation updates due to profile-analysis being combined

### PR DESCRIPTION
**Pull request by nv-braf**
Updated CLI, config, and quick start documentation now that profile & analysis are combined.
I have still left the documentation for analysis in, but have indicated that it's usage has been combined with profile and we will be deprecating its usage in the near future.

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -34,39 +34,51 @@ in which it is being run. Currently model analyzer supports 2 modes.
...
-online mode, the best model configuration will be the one that maximizes 
-throughput. If a latency budget is specified to the [analyze subcommand](#subcommand-analyze) via 
-`--latency-budget`, then the best model configuration will be the one with the highest throughput in the given budget. 
+the optimal model configuration for an online inference scenario. By default in
+online mode, the best model configuration will be the one that maximizes
+throughput. If a latency budget is specified to the [profile subcommand](#subcommand-profile) via
+`--latency-budget`, then the best model configuration will be the one with the highest throughput in the given budget.
 
-In online mode the analyze and report subcommands will generate summaries specific to online inference. 
+In online mode the analyze and report subcommands will generate summaries specific to online inference.
```
Comment (tgerdesnv) [inline docs/cli.md:42]: doesn't profile do this now as well?

File: docs/cli.md [hunk 2]
```diff
@@ -34,39 +34,51 @@ in which it is being run. Currently model analyzer supports 2 modes.
 ### Online Mode
 
 This is the default mode. When in this mode, Model Analyzer will operate to find
-the optimal model configuration for an online inference scenario. By default in 
-online mode, the best model configuration will be the one that maximizes 
-throughput. If a latency budget is specified to the [analyze subcommand](#subcommand-analyze) via 
-`--latency-budget`, then the best model configuration will be the one with the highest throughput in the given budget. 
+the optimal model configuration for an online inference scenario. By default in
+online mode, the best model configuration will be the one that maximizes
+throughput. If a latency budget is specified to the [profile subcommand](#subcommand-profile) via
```
Comment (tgerdesnv) [inline docs/cli.md:39]: do we want to say 'profile or analyze' for cases like this? Or only speak of profile?

File: docs/cli.md [hunk 3]
```diff
@@ -34,39 +34,51 @@ in which it is being run. Currently model analyzer supports 2 modes.
...
-The `profile` subcommand allows the user to run model inferences using perf
-analyzer, and collect metrics like throughput, latency and memory usage. Use the
-following command to see the usage and argument descriptions for the subcommand.
+The `profile` subcommand begins by loading the "latest" checkpoint (if available) in
+the checkpoint directory. It will then run model inferences using perf
+analyzer, and collect metrics like throughput, latency and memory usage for
+any measurements not present in the checkpoint.
+
+Next, it sorts the models specified in the CLI or
+config YAML, using the objectives specified in the config YAML. Finally, it constructs summary PDFs
```
Comment (tgerdesnv) [inline docs/cli.md:73]: the objectives -> any objectives 
(or else it implies that it is required)

File: docs/cli.md [hunk 4]
```diff
@@ -91,68 +103,96 @@ model, the Model Analyzer will save the measurements it has collected into the
...
-  ```
-  $ model-analyzer profile -m /home/model_repo --output-model-repository-path=/home/output_repo --profile-models resnet50_libtorch
-  ```
+```
+$ model-analyzer profile -m /home/model_repo --output-model-repository-path=/home/output_repo --profile-models resnet50_libtorch
+```
 
 4. Run profile over manually defined configurations for a models `classification_malaria_v1` and `classification_chestxray_v1` located in `/home/model_repo` using the YAML config file
 
-  ```
```
Comment (tgerdesnv) [inline docs/cli.md:117]: did you mean to remove this?

File: docs/cli.md [hunk 5]
```diff
@@ -91,68 +103,96 @@ model, the Model Analyzer will save the measurements it has collected into the
...
 
-profile_models: 
+profile_models:
   classification_malaria_v1:
     model_config_parameters:
       instance_group:
-        -
-          kind: KIND_GPU
-          count: [1,2]
+        - kind: KIND_GPU
```
Comment (tgerdesnv) [inline docs/cli.md:143]: i'm not 100% sure that this is equivalent yaml

File: docs/cli.md [hunk 6]
```diff
@@ -91,68 +103,96 @@ model, the Model Analyzer will save the measurements it has collected into the
...
+  vgg16_graphdef:
+    objectives:
+      - gpu_used_memory
+    constraints:
+      perf_latency_p99:
+        max: 15
+```
+
+```
+$ model-analyzer profile -f /path/to/config.yaml
```
Comment (tgerdesnv) [inline docs/cli.md:184]: duplicated command from above?

**Review by tgerdesnv — COMMENTED** [review]
I think the bottom of checkpoints.md needs a tweak as well

File: docs/cli.md (hunk 1 above)
Comment (nv-braf) [reply docs/cli.md:42]: I switched this to say `"profile and report subcommands"`

File: docs/cli.md (hunk 2 above)
Comment (nv-braf) [reply docs/cli.md:39]: I only want to speak of profile. With the eventual goal of removing analyze altogether.

File: docs/cli.md (hunk 3 above)
Comment (nv-braf) [reply docs/cli.md:73]: Done

File: docs/cli.md (hunk 6 above)
Comment (nv-braf) [reply docs/cli.md:184]: Copy/Paste error. Removed

File: docs/cli.md (hunk 4 above)
Comment (nv-braf) [reply docs/cli.md:117]: Nope. That was a copy/paste error.

**Comment by nv-braf** [comment]
> I think the bottom of checkpoints.md needs a tweak as well

Which part? I don't see any mention of analysis in that file

**Comment by tgerdesnv** [comment]
> > I think the bottom of checkpoints.md needs a tweak as well
> 
> Which part? I don't see any mention of analysis in that file

"The checkpoint directory should be removed between consecutive runs of the model-analyzer profile command."

File: docs/cli.md (hunk 5 above)
Comment (nv-braf) [reply docs/cli.md:147]: I tried it out (in a unit test) and it worked fine

Comment (tgerdesnv) [reply docs/cli.md:147]: fair enough

**Review by tgerdesnv — APPROVED** [review]

**Pull request merged**